code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1571,8 +1571,8 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 //////////
 
 /datum/species/proc/handle_fire(mob/living/carbon/human/H, no_protection = FALSE)
-	if(H.has_trait(TRAIT_NOFIRE))
-		return
+	if(!CanIgniteMob(H))
```

Comment:
no change in behavior with this, it just wasn't using the proc right below that checks the same thing.